ref/Microsoft.Build.Tasks.Core/net/Microsoft.Build.Tasks.Core.cs

```diff
@@ -1281,6 +1293,15 @@ public partial class XslTransformation : Microsoft.Build.Tasks.TaskExtension
         public Microsoft.Build.Framework.ITaskItem XslInputPath { get { throw null; } set { } }
         public override bool Execute() { throw null; }
     }
+    public sealed partial class ZipDirectory : Microsoft.Build.Tasks.TaskExtension
+    {
+        public ZipDirectory() { }
+        [Microsoft.Build.Framework.RequiredAttribute]
+        public Microsoft.Build.Framework.ITaskItem DestinationFile { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
+        [Microsoft.Build.Framework.RequiredAttribute]
+        public Microsoft.Build.Framework.ITaskItem SourceDirectory { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
```

Comment:
I'm a bit worried about this interface. It's pretty much impossible to get correct incremental behavior out of this. You'd have to:

* Enumerate the contents of the folder _just before_ the target that invokes `ZipDirectory`.
* Additionally compute a hash of those items to avoid the removed-an-input problem.
* Use the new stuff in the target that calls this.

Accepting `Item[]` would be more complex in implementation but easier to make correct.

Reply:
I discussed this with @AndyGerlicher and we're concerned more about the complexity required to make it incremental out-of-the-box.  So we're going to leave it as-is.

Reply:
I think if we wrote a target that used this we would need to make it incremental. But given that it zips a directory, it doesn't seem worth the complexity to implement at the task level. It might be worth writing a sample in the docs page that is a target that zips your output folder that would be incremental as an example.